src/simulate.jl

```diff
@@ -67,9 +67,8 @@ function simulate(parameters, biomass; concentration::Vector{Float64}=rand(Float
 
   affect_function = parameters[:rewire_method] == :none ? remove_species! : remove_species_and_rewire!
 
-  extinction_callback = DiscreteCallback(species_under_extinction_threshold, affect_function)
-  sol = solve(prob, alg, callback = extinction_callback, saveat=t_keep, dense=false, save_timeseries=false)
-
+  extinction_callback = DiscreteCallback(species_under_extinction_threshold, affect_function; save_positions=(true,true))
+  sol = solve(prob, alg, callback = CallbackSet(extinction_callback, PositiveDomain()), saveat=t_keep, dense=false, save_timeseries=false, force_dtmin=true)
```

Comment:
I'm starting to get the sense that there are *many* things wrong with the model, but that's the best food web dynamics model we have.

Reply:
Why do you need it? What's going on that's causing it? What's going to infinity or diving by zero or ?

Reply:
When it's removed, the `PositiveDomain()` callback complains (and the solver terminates). It seems to make very little difference in the output.

Reply:
Abundences can go below zero by the amount of `abstol`. That's a Shampine thing. But your model is okay with continuing after a dtmin step? It doesn't end up taking billions of small steps? I wonder if this is https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/issues/273

Reply:
Ahh, there could be some issue in the callback since I got this issue I forgot about: https://github.com/JuliaDiffEq/DiffEqCallbacks.jl/issues/44

Reply:
Thank for the clarification about abstol. So the "correct" way to set these species to 0 would be to use `abstol` [here](https://github.com/PoisotLab/BioEnergeticFoodWebs.jl/blob/next-speed/src/simulate.jl#L54)?

Reply:
The strange thing (from my point of view) is that the model warns about being unable to restrict values to domain maybe once or twice for 500 timesteps, so this seems to be really rare.

Reply:
For now `isoutofdomain=(u,p,t)->any(x->x<0,u)` is the safest way to ensure positivity until I find out why that dtmin issue is popping up. With that, since it cannot hit zero, I would keep the extinction check above zero. `10eps()` is essentially zero though. I would take something like `1000eps()` or `1e5eps()` as a threshold to zero since I'm sure most people aren't solving this equation with like `abstol=reltol=1e-12`.

Reply:
That's a fair assumption. I think the original papers from 2005-ish used 1e-10, but no one is sure because the code got lost. I'll try this later today and see whether that works. Thanks!